src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityService.kt

```diff
@@ -0,0 +1,104 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchPrisoner
+import java.time.Clock
+import java.time.LocalDate
+
+@Service
+class EligibilityService(
+  private val prisonApiClient: PrisonApiClient,
+  private val clock: Clock,
+) {
+
```

Comment:
Used && chaining logic due to the short circuit nature of the && call.

Reply:
would be tempted to use extension functions for these... would add a lot of uniformity.

The nice thing is you'd just have to remove the parameters from the functions and whack a `PrisonerSearchPrisoner.` to the method and it should all still compile